ert/data/record/_transformation.py

```diff
@@ -227,3 +229,67 @@ async def _save_record_to_file(record: Record, location: Path, mime: str) -> Non
     else:
         async with aiofiles.open(str(location), mode="wb") as fb:
             await fb.write(record.data)  # type: ignore
+
+
+class RecordTreeTransformation(RecordTransformation):
+    """
+    Write all leaf records of a NumericalRecordTree to individual files.
+    """
+
+    def __init__(self, sub_path: Optional[str] = None):
+        if sub_path is not None:
+            raise NotImplementedError
+
+    async def transform_input(
+        self,
+        record: Record,
+        mime: str,
+        runpath: Path,
+        location: Path,
+    ) -> None:
+        if not isinstance(record, NumericalRecordTree):
+            raise TypeError("Only NumericalRecordTrees can be transformed.")
+        for key, leaf_record in record._flat_record_dict.items():
+            location_key = f"{key}-{location}"
+            await get_serializer(mime).encode_to_path(
+                leaf_record.data, path=runpath / location_key
+            )
+
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        raise NotImplementedError
+
+
+class EclSumTransformation(RecordTreeTransformation):
+    """A transformation that will transform binary output from Eclipse into a
+    NumericalRecordTree."""
+
+    def __init__(self, smry_keys: List[str]):
```

Comment:
Is the empty list something to worry about here? Because execution of this transformation on the empty list of keys will blow up if I am reading this correctly.

Since transformations are likely to happen on the cluster, we should strive to fail early.

Reply:
Added ValueError if empty list is supplied.